integrations/fluid.py

```diff
@@ -56,14 +53,26 @@ def get_participants(self) -> list:
         return participants
     
     def get_relevant_vaults(self, block: int) -> list:
-        vaults = []
-        totalVaults = call_with_retry(vaultFactory_contract.functions.totalVaults(), block)
-        for i in range(1, totalVaults + 1):
-            vaultAddress = call_with_retry(vaultFactory_contract.functions.getVaultAddress(i), block)
+        if block in self.blocknumber_to_susdeVaults:
+            return self.blocknumber_to_susdeVaults[block]
+        
+        
+        if self.blocknumber_to_susdeVaults != {}:
```

Comment:
this could be better performant: 
1. getAllVaultsAddresses() 
2. check if count increased since last sync (store this somewhere locally)
3. if increased, continue with existing logic 

or alternatively store total vaults at last sync so we dont have to fetch it at a previous block height?